src/hotspot/share/jvmci/jvmci.hpp

```diff
@@ -104,10 +112,19 @@ class JVMCI : public AllStatic {
      first_permanent_bailout = code_too_large
   };
 
+  // Returns true iff JVMCIThreadsPerNativeLibraryRuntime == 0.
+  static bool using_singleton_shared_library_runtime() {
+    return JVMCIThreadsPerNativeLibraryRuntime == 0;
+  }
+
+  // Returns true iff there is a new shared library JavaVM per compilation.
+  static bool one_shared_library_javavm_per_compilation() {
+    return JVMCIThreadsPerNativeLibraryRuntime == 1 && JVMCICompilerIdleDelay == 0;
```

Comment:
Why does this check for `JVMCICompilerIdleDelay == 0` and where is this method used?

Reply:
It's used in `JVMCIRuntime::post_compile`.

The combination of `JVMCIThreadsPerNativeLibraryRuntime == 1 && JVMCICompilerIdleDelay == 0` is the stress test mode for multi-isolate support. It forces a new libgraal isolate to be used for every single compilation. I've updated the help text for `JVMCICompilerIdleDelay ` to document this mode: https://github.com/openjdk/jdk/pull/8262/commits/c6d93f0d597230f68b5fc37add874ecf1437e55f